packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -24,9 +24,31 @@ import 'package:flutter/scheduler.dart';
 
 import 'basic.dart';
 import 'framework.dart';
+import 'layout_builder.dart';
 import 'lookup_boundary.dart';
 import 'ticker_provider.dart';
 
+/// The signature of the widget builder callback used in
+/// [OverlayPortal.overlayChildLayoutBuilder].
+typedef OverlayChildLayoutBuilder =
+    Widget Function(BuildContext context, OverlayChildLayoutInfo info);
+
+/// The additional layout information available to the
+/// [OverlayPortal.overlayChildLayoutBuilder] callback.
+extension type OverlayChildLayoutInfo._(
```

Comment:
:)

Personally, I would have gone with a regular class here for slightly better readability and ergonomics. But this is fine.

Reply:
That way I don't have to manually write the `==` and `hashCode` implementation.